ext/storage_sources/s3_store/s3_connection.cpp

```diff
@@ -118,14 +129,18 @@ S3Connection::PutObject(const std::string &objectKey, const std::string &fileNam
     request.SetBody(inputData);
 
     Aws::S3Crt::Model::PutObjectOutcome outcome = _s3CrtClient.PutObject(request);
-    if (outcome.IsSuccess()) {
+    Aws::Http::HttpResponseCode resCode = outcome.GetError().GetResponseCode();
```

Comment:
Same as before, limit scope of resCode to the failure block. This comment holds for other places as well.